src/Sentry/SentryClient.cs

```diff
@@ -431,11 +465,12 @@ private bool CaptureEnvelope(Envelope envelope)
     /// <summary>
     /// Disposes this client
     /// </summary>
-    /// <inheritdoc />
     public void Dispose()
     {
         _options.LogDebug("Flushing SentryClient.");
 
+        Metrics.FlushAsync().GetAwaiter().GetResult();
```

Comment:
If this throws (e.g: OperationCancelled) the worker task below won't complete.

I think best way here is:

// start tasks
var metricFlushTask = Metrics.FlushAsync()
var workerFlushTask = Worker.FlushAsync(_options.ShutdownTimeout)
try
Task.WhenAll(metricsFlushTask, wrokerFlushTask);
catch
catch anything here and log.